### PR TITLE
manifest: openthread: Regular OpenThread upmerge

### PR DESCRIPTION
**Pull request by maciejbaczmanski**
Update OpenThread up to `2aeb8b8` and align KConfig options.

Additionally add `OPENTHREAD_STORE_FRAME_COUNTER_AHEAD` Kconfig option
and set it to 100000, as after calculations it appears to
be a more suitable value.

### CODE REVIEW

File: modules/openthread/Kconfig.thread [hunk 1]
```diff
@@ -204,3 +204,9 @@ config OPENTHREAD_NAT64_CIDR
 	  to set source address of the outgoing translated IPv4 packets.
 	  The CIDR must have four bytes in the address with the
 	  non-zero length of prefix (e.g., "127.0.0.1/24").
+
+config OPENTHREAD_STORE_FRAME_COUNTER_AHEAD
+	int "Openthread value ahead of the current frame counter for persistent storage."
+	default 100000
+	help
+	   Openthread value ahead of the current frame counter for persistent storage.
```
Comment (pdgendt) [inline modules/openthread/Kconfig.thread:212]: Nit, although having the same value for the prompt and description doesn't seem helpful:

```suggestion
	  Openthread value ahead of the current frame counter for persistent storage.
```

File: modules/openthread/Kconfig.features [hunk 2]
```diff
@@ -158,6 +164,9 @@ config OPENTHREAD_DUA
 	help
 	  Enable Domain Unicast Address feature for Thread 1.2
 
+config OPENTHREAD_DYNAMIC_STORE_FRAME_AHEAD_COUNTER
+	bool "Dynamic store frame ahead countert"
```
Comment (pdgendt) [inline modules/openthread/Kconfig.features:168]:
```suggestion
	bool "Dynamic store frame ahead counter"
```

File: modules/openthread/Kconfig.thread (hunk 1 above)
Comment (maciejbaczmanski) [reply modules/openthread/Kconfig.thread]: Thanks, fixed in other places as well.
I've shortened the prompt, leaving the original description the same as OpenThread's description

**Review by canisLupus1313 — DISMISSED** [review]

**Review by pdgendt — DISMISSED** [review]

File: modules/openthread/Kconfig.features [hunk 3]
```diff
@@ -52,6 +52,9 @@ config OPENTHREAD_BORDER_AGENT
 config OPENTHREAD_BORDER_AGENT_EPHEMERAL_KEY_ENABLE
 	bool "Border agent ephemeral PSKc"
 
+config OPENTHREAD_BORDER_AGENT_ID
+	bool "Dreate and save border agent ID"
```
Comment (edmont) [inline modules/openthread/Kconfig.features:56]:
```suggestion
	bool "Create and save border agent ID"
```

**Review by edmont — DISMISSED** [review]

File: modules/openthread/Kconfig.features (hunk 3 above)
Comment (maciejbaczmanski) [reply modules/openthread/Kconfig.features]: Thanks! fixed

**Review by edmont — DISMISSED** [review]

**Review by pdgendt — DISMISSED** [review]

**Review by markaj-nordic — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**